lib/src/main/java/edu/stanford/futuredata/macrobase/operator/Operator.java

```diff
@@ -1,5 +1,13 @@
 package edu.stanford.futuredata.macrobase.operator;
 
+/**
+ * Generic interface for operations. Should not modify input.
```

Comment:
Most transforms I can think of are going to modify a significant fraction of the column values, in which case creating a new column isn't much overhead on top of computing these values.

If we come across an operation that really benefits from in-place mutation I figure we can create a new "Mutator" interface to make the distinction very explicit.